it/v1/endpoints/RetrieveEmploymentsExpensesControllerISpec.scala

```diff
@@ -16,267 +16,69 @@
-    }
-
-    def errorBody(code: String): String =
-      s"""
-         |{
-         |  "failures": [
-         |    {
-         |      "code": "$code",
-         |      "reason": "if message"
-         |    }
-         |  ]
-         |}
-    """.stripMargin
-
-  }
-
   "Calling the retrieve endpoint" should {
-
     "return a 200 status code" when {
+      "valid latest request is made" in new NonTysTest {
+        override def setupStubs(): Unit = {
+          DownstreamStub.onSuccess(DownstreamStub.GET, downstreamUri, Map("view" -> latestSourceDownstream), Status.OK, downstreamResponseJsonLatest)
+        }
 
-      "valid latest request is made" in new Test {
+        val response: WSResponse = await(request(latestMtdUri).get())
+        response.status shouldBe Status.OK
+        response.json shouldBe mtdResponseWithHateoasLinksLatest()
+        response.header("X-CorrelationId").nonEmpty shouldBe true
+        response.header("Content-Type") shouldBe Some("application/json")
+      }
 
-        override def setupStubs(): StubMapping = {
-          AuditStub.audit()
-          AuthStub.authorised()
-          MtdIdLookupStub.ninoFound(nino)
-          DownstreamStub.onSuccess(DownstreamStub.GET, ifsUri, Map("view" -> desSource), Status.OK, desResponseBody)
+      "valid hmrcHeld request is made" in new NonTysTest {
+        override def setupStubs(): Unit = {
+          DownstreamStub.onSuccess(
+            DownstreamStub.GET,
+            downstreamUri,
+            Map("view" -> hmrcHeldSourceDownstream),
+            Status.OK,
+            downstreamResponseJsonHmrcHeld)
         }
 
-        val response: WSResponse = await(request(latestUri).get())
+        val response: WSResponse = await(request(hmrcHeldMtdUri).get())
         response.status shouldBe Status.OK
-        response.json shouldBe latestResponseBody
+        response.json shouldBe mtdResponseWithHateoasLinksHmrcHeld
         response.header("X-CorrelationId").nonEmpty shouldBe true
         response.header("Content-Type") shouldBe Some("application/json")
       }
-      "valid hmrcHeld request is made" in new Test {
 
-        override def setupStubs(): StubMapping = {
-          AuditStub.audit()
-          AuthStub.authorised()
-          MtdIdLookupStub.ninoFound(nino)
-          DownstreamStub.onSuccess(DownstreamStub.GET, ifsUri, Map("view" -> hmrcHeldDesSource), Status.OK, hmrcHeldDesResponseBody)
+      "valid user request is made" in new NonTysTest {
+        override def setupStubs(): Unit = {
+          DownstreamStub.onSuccess(DownstreamStub.GET, downstreamUri, Map("view" -> userSourceDownstream), Status.OK, downstreamResponseJsonCustomer)
         }
 
-        val response: WSResponse = await(request(hmrcHeldUri).get())
+        val response: WSResponse = await(request(userMtdUri).get())
         response.status shouldBe Status.OK
-        response.json shouldBe hmrcHeldResponseBody
+        response.json shouldBe mtdResponseWithHateoasLinksUser
```

Comment:
Just out of interest, is the value of having a fixtures file with these responses in to make the ISpec less cluttered with Json response bodies etc?

Reply:
It's to make all the tests less cluttered, the Fixtures are used in a lot of the unit tests too. You can tell from the number of lines removed vs lines added how much has been deleted!